pandas/core/arrays/datetimes.py

```diff
@@ -1933,7 +1933,13 @@ def sequence_to_dt64ns(
             data, inferred_tz = objects_to_datetime64ns(
                 data, dayfirst=dayfirst, yearfirst=yearfirst
             )
-            tz = _maybe_infer_tz(tz, inferred_tz)
+            if tz and inferred_tz:
```

Comment:
I think this would change e.g. DatetimeIndex behavior:

```
dti = pd.date_range("2016-01-01", periods=3, tz="UTC")
pd.DatetimeIndex(dti, tz="US/Pacific")   # <-- raises in master
```

does that still raise in this PR?

Reply:
I think this path is *only* hit if its object in the first place; so your example bypasses

Reply:
To confirm jreback, in this PR the error is still raised: `TypeError: data is already tz-aware UTC, unable to set specified tz: US/Pacific`

Reply:
makes sense, thanks